spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java

```diff
@@ -0,0 +1,463 @@
+          .enableDynamicOverride(true)
+          .description(
+              "When enabled, operator would trim state transition history when a "
+                  + "new attempt starts, keeping previous attempt summary only.")
+          .typeParameterClass(Boolean.class)
+          .defaultValue(true)
+          .build();
+
+  public static final ConfigOption<String> SPARK_APP_STATUS_LISTENER_CLASS_NAMES =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.reconciler.appStatusListenerClassNames")
+          .enableDynamicOverride(false)
+          .description("Comma-separated names of SparkAppStatusListener class implementations")
+          .typeParameterClass(String.class)
+          .defaultValue("")
+          .build();
+
+  public static final ConfigOption<Boolean> DYNAMIC_CONFIG_ENABLED =
+      ConfigOption.<Boolean>builder()
+          .key("spark.kubernetes.operator.dynamicConfig.enabled")
+          .enableDynamicOverride(false)
+          .description(
+              "When enabled, operator would use config map as source of truth for config "
+                  + "property override. The config map need to be created in "
+                  + "spark.kubernetes.operator.namespace, and labeled with operator name.")
+          .typeParameterClass(Boolean.class)
+          .defaultValue(false)
+          .build();
+
+  public static final ConfigOption<String> DYNAMIC_CONFIG_SELECTOR =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.dynamicConfig.selector")
+          .enableDynamicOverride(false)
+          .description("The selector str applied to dynamic config map.")
+          .typeParameterClass(String.class)
+          .defaultValue(Utils.labelsAsStr(Utils.defaultOperatorConfigLabels()))
+          .build();
+
+  public static final ConfigOption<Integer> DYNAMIC_CONFIG_RECONCILER_PARALLELISM =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.dynamicConfig.reconcilerParallelism")
+          .enableDynamicOverride(false)
+          .description(
+              "Parallelism for dynamic config reconciler. Unbounded pool would be used "
+                  + "if set to non-positive number.")
+          .typeParameterClass(Integer.class)
+          .defaultValue(1)
+          .build();
+
+  public static final ConfigOption<Integer> RATE_LIMITER_REFRESH_PERIOD_SECONDS =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.rateLimiter.refreshPeriodSeconds")
+          .enableDynamicOverride(false)
+          .description("Operator rate limiter refresh period(in seconds) for each resource.")
+          .typeParameterClass(Integer.class)
+          .defaultValue(15)
+          .build();
+
+  public static final ConfigOption<Integer> RATE_LIMITER_LIMIT_FOR_PERIOD =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.rateLimiter.limitForPeriod")
```

Comment:
Since `rateLimiter` already includes `limit`, can we be more specific?
For example, `limitForPeriod` -> `maxLoopForPeriod`? WDYT?